### PR TITLE
feat: add regional twr header in the access token

### PR DESCRIPTION
**Pull request by charliesantos**
# Feature #

This PR adds `AccessTokenOptions.region` parameter to support regional authentication. This value will be encoded in the `twr` header of the JWT. If the parameter is not specified or invalid, no `twr` value is added. See this [document](https://docs.google.com/document/d/17HVcWbfysfhrNNEfewO_bnQaDzVTiOw7Mjeg5ITTLCY/edit#heading=h.ykvt29hsgqdp) for more details.
JIRA: https://issues.corp.twilio.com/browse/CLIENT-8142

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar) [N/A]
- [x] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-node/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file [N/A]
- [x] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: spec/unit/jwt/AccessToken.spec.js [hunk 1]
```diff
@@ -35,6 +35,27 @@ describe('AccessToken', function() {
...
+
+      it('should add twr header when region is provided', function() {
+        var token = new twilio.jwt.AccessToken(accountSid, keySid, secret, {region: 'foo'});
+        var decoded = jwt.decode(token.toJwt(), {complete: true});
+
+        expect(decoded.header.twr).toBe('foo');
+      });
+
+      ['', undefined, null, {}, 1, 0].forEach(function(value) {
+        it('should not twr header if region is ' + value, function() {
```
Comment (eshanholtz) [inline spec/unit/jwt/AccessToken.spec.js:50]: nit: "should not *add* twr..."

File: spec/unit/jwt/AccessToken.spec.js [hunk 2]
```diff
@@ -46,7 +67,7 @@ describe('AccessToken', function() {
       });
     });
 
-    it('should accept different algorithsm', function() {
+    it('should accept different algorithm', function() {
```
Comment (eshanholtz) [inline spec/unit/jwt/AccessToken.spec.js:70]: nit: algorithm*s*

File: lib/jwt/AccessToken.js [hunk 3]
```diff
@@ -264,11 +266,17 @@ _.extend(AccessToken.prototype, {
     };
     if (_.isNumber(this.nbf)) { payload.nbf = this.nbf; }
 
+    var header = {
+      cty: 'twilio-fpa;v=1',
+      typ: 'JWT'
+    };
+
+    if (this.region && typeof this.region === 'string') {
```
Comment (eshanholtz) [inline lib/jwt/AccessToken.js:274]: While we still have a lodash as a dependency, I'd prefer the second half of this statement be `_.isString(this.region)` for consistency

**Review by eshanholtz — APPROVED** [review]

File: lib/jwt/AccessToken.js [hunk 4]
```diff
@@ -208,6 +208,7 @@ _.extend(VoiceGrant.prototype, {
  * @param {number} [options.ttl=3600] - Time to live in seconds
  * @param {string} [options.identity] - The identity of the first person
  * @param {number} [options.nbf] - Time from epoch in seconds for not before value
+ * @param {string} [options.region] - Home region to use
```
Comment (idelgado) [inline lib/jwt/AccessToken.js:211]: > The region value associated with this account

File: lib/jwt/AccessToken.d.ts [hunk 5]
```diff
@@ -158,6 +158,10 @@ declare namespace AccessToken {
      * Time from epoch in seconds for not before value
      */
     nbf?: number;
+    /**
+     * Home region to use
```
Comment (idelgado) [inline lib/jwt/AccessToken.d.ts:162]: I think externally we don't use the term `Home`.
> The region value associated with this account

**Review by idelgado — CHANGES_REQUESTED** [review]

**Review by idelgado — APPROVED** [review]

**Pull request merged**